tests/robustness/multiclient/client/client.go

```diff
@@ -0,0 +1,70 @@
+// +build darwin,amd64 linux,amd64
+
+// Package client manages client specific info
+package client
+
+import (
+	"context"
+	"sync"
+
+	petname "github.com/dustinkirkland/golang-petname"
+	"github.com/google/uuid"
+)
+
+type key int
+
+const (
+	clientKey key = 0
+	nameLen   int = 2
+)
+
+// Client is a unique client for use in multiclient robustness tests.
+type Client struct {
+	ID string
+}
+
+func newClient() *Client {
+	petname.NonDeterministicMode()
+
+	return &Client{
+		ID: petname.Generate(nameLen, "-") + "-" + uuid.NewString(),
+	}
+}
+
+// WrapContext returns a copy of ctx with a new client.
+func WrapContext(ctx context.Context) context.Context {
```

Comment:
Since this is an (unusual) constructor, I suggest you name it properly.  I suggest `NewClientContext` or `ContextWithNewClient`. 

Ideally you should have a separate constructor for `Client` and then a helper method to wrap it in the context.